src/build.py

```diff
@@ -1416,20 +1416,40 @@ def ValidateLLVMTorture(indir, ext, opt):
 
 class Build(object):
     def __init__(self, name_, runnable_, os_filter=None, is_default=True,
-                 *args, **kwargs):
+                 incremental_build_dir=None, *args, **kwargs):
         self.name = name_
         self.runnable = runnable_
         self.os_filter = os_filter
-        self.is_deafult = is_default
+        self.incremental_build_dir = incremental_build_dir
         self.args = args
         self.kwargs = kwargs
 
+        if incremental_build_dir:
+            self.kwargs['build_dir'] = incremental_build_dir
+
     def Run(self):
         if self.os_filter and not self.os_filter.Check(BuilderPlatformName()):
             print("Skipping %s: Doesn't work on %s" %
                   (self.runnable.__name__, BuilderPlatformName()))
             return
-        self.runnable(*self.args, **self.kwargs)
+
+        try:
+            # When using LTO we always want a clean build (the previous
+            # build was non-LTO)
+            if self.incremental_build_dir and options.use_lto:
+                RemoveIfBot(self.incremental_build_dir)
```

Comment:
can the if be outside the try?

Reply:
Yeah, that would make sense given that the except repeats it :)